lib/service/Service.ts

```diff
@@ -196,7 +196,17 @@ class Service extends EventEmitter {
     const { orderId, pairId, peerPubKey } = args;
     const quantity = args.quantity > 0 ? args.quantity : undefined; // passing 0 quantity will work fine, but it's prone to bugs
 
-    return this.orderBook.executeSwap(orderId, pairId, peerPubKey, quantity);
+    const maker = this.orderBook.removePeerOrder(orderId, pairId, peerPubKey, quantity).order;
```

Comment:
I don't fully understand why `removePeerOrder` and `stampOwnOrder` was moved to the service level. Can you please elaborate?

Reply:
Basically I tried to make the `executeSwap` method in OrderBook.ts accept only two orders as parameters so it can be called internally as well as from Service.ts. Service.ts gets the orders to pass to `executeSwap` by calling these two methods. If it's not very clear I can always restructure it or add some comments.

Reply:
Fair enough. I think it makes sense this way since `removePeerOrder` and `stampOwnOrder` are public. I actually made `stampOwnOrder` private in #667, but I'll revert the changes.